---
title: "C18"
author: "Hongji Jiang"
date: "11/26/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
library(GenomicDataCommons)
library(TCGAbiolinks)
library(maftools)
```

```{r}
status()
```

```{r}
projects <- getGDCprojects()
head(projects)
```

```{r}
cases_by_project <- cases() %>%
  facet("project.project_id") %>%
  aggregations()
head(cases_by_project)
```
```{r}
x <- cases_by_project$project.project_id

# Make a custom color vector for our plot
colvec <- rep("lightblue", nrow(x))
colvec[(x$key == "TCGA-PAAD")] <- "red"
colvec
```

```{r}
barplot(x$doc_count, col = colvec,log = "y", names.arg = x$key, las =2)
```


```{r}
samp <- getSampleFilesSummary("TCGA-PAAD")
head(samp)
```


```{r}
query <- GDCquery(project="TCGA-PAAD",
                  data.category="Transcriptome Profiling",
                  data.type="Gene Expression Quantification")

ans <- getResults(query)
```

```{r}
library(bio3d)
seqs <- read.fasta("lecture18_sequences.fa")
seqs
```

```{r}
ide <- conserv(seqs$ali, method="identity")
mutant.sites <- which(ide < 1) 
gaps <- gap.inspect(seqs)
mutant.sites <- mutant.sites[mutant.sites %in% gaps$f.inds]

mutant.sites
```

```{r}
mutant.names <- paste0(seqs$ali["P53_wt",mutant.sites],
                       mutant.sites,
                       seqs$ali["P53_mutant",mutant.sites])

mutant.names
```

```{r}
start.position <- mutant.sites - 8
end.position <-  mutant.sites + 8

# Blank matrix to store sub-sequences
store.seqs <- matrix("-", nrow=length(mutant.sites), ncol=17)
rownames(store.seqs) <- mutant.names

## Extract each sub-sequence
for(i in 1:length(mutant.sites)) {
  store.seqs[i,] <- seqs$ali["P53_mutant",start.position[i]:end.position[i]]
}

store.seqs
```

```{r}
write.fasta(seqs=store.seqs, ids=mutant.names, file="subsequences.fa")
```

```{r}
wt <- get.seq("P01116")

## Here we make four mutants namely: G12V, Q22N, T74S and A130V
mutant <- wt
mutant$ali[ c(12,22,74,130)] <- c("V", "N", "S", "V")

write.fasta( seqbind(wt, mutant), ids=c("wt","mutant"), file="kras-sequences.fa")
```

